middleware.js

```diff
@@ -182,9 +182,22 @@ module.exports = function(compiler, options) {
 
 	webpackDevMiddleware.getFilenameFromUrl = getFilenameFromUrl;
 
-	webpackDevMiddleware.invalidate = function() {
-		if(watching) watching.invalidate();
+	webpackDevMiddleware.waitUntilValid = function(callback) {
+		callback = callback || function(){};
+		if (!watching || !watching.running) callback();
+		else callbacks.push(callback);
 	};
```

Comment:
Could you make that `waitUntilValid` shares code with the `ready` function which is internally used.